### DISCUSSION TITLE
How can I implement theia extension using vue instead of react?

### DISCUSSION THREAD

**Discussion by hou-kx** [eclipse-theia/theia#12862] [Q&A] [answered]
In order to reuse some of the vue components I've implemented before, I'd like to use vue for theia's front-end extension instead of the official react approach.

 I would like to ask if there are any experts who have successfully implemented vue to build theia's front-end pages.

**Comment by msujew** [comment, marked as answer]
Hey @hou-kx,

I'm not sure whether anyone has done that yet, but it should be fairly simple, assuming Vue has a similar portalling mechanism as react.

You need some mechanism to render a vue component into a native HTML element. Every Theia widget is based on a single HTML node. We do that in our own react widgets via `React.createRoot` [here](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/widgets/react-widget.tsx). The component then needs to re-render on every `onUpdateRequest` of the `BaseWidget`.

**Reply by hou-kx** [reply]
OK, well then, this way has worked before (by implementing VueWidget). And I was thinking of finding more convenient ways to do it, thanks anyway!

**Reply by woaixiaobo** [reply]
I succeeded in using the functional writing method of vue, but failed to use the template compilation method. Would you like to ask if you realized it through the template compilation method?
